pilot/pkg/status/resourcelock_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"sync/atomic"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestResourceLock_Lock(t *testing.T) {
```

Comment:
this test is flaky: `124 runs so far, 13 failures (91% pass rate). 8.873057408s avg, 15.261575091s max, 2.147483647s min`

Can you improve it to not rely on precise timings of time.sleep, possibly with retry.UntilSuccess?